### PR TITLE
feat(2605) Option to display all screwdriver jobs in a chronological order

### PR DESCRIPTION
**Pull request by adong**
## Context

image

## Objective
Provide an option for user to set event display at pipeline level, whether is _grouped events_ or not

## References

- https://github.com/screwdriver-cd/screwdriver/issues/2605
- https://github.com/screwdriver-cd/ui/pull/658/files

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: app/components/pipeline-events-list/component.js [hunk 1]
```diff
@@ -7,7 +7,30 @@ import moment from 'moment';
...
   errorMessage: '',
+  isGroupedEvents: computed('pipeline', {
+    get() {
+      const isGroupedEvents = getWithDefault(
+        this.pipeline,
+        'settings.groupedEvents',
+        false
+      );
+
+      console.log('isGroupedEvents', isGroupedEvents);
```
Comment (klu909) [inline app/components/pipeline-events-list/component.js:20]: remove console.log

File: app/components/pipeline-options/template.hbs [hunk 2]
```diff
@@ -186,6 +186,31 @@
...
 
+  <div class="col-xs-12 col-md-8">
+    <section class="preference">
+      <h3>Pipeline Preference</h3>
+      <ul>
+        <li>
+          <div class="row">
+            <div class="col-xs-10">
+              <h4>Group Events</h4>
+              <p>Setup your pipeline preference to group events or not</p>
```
Comment (jithine) [inline app/components/pipeline-options/template.hbs:197]: we would need to have an explanation for what `grouping` is as `title` or something.
Something to the effect `When grouping is enabled events which share a common ancestor will be grouped together`

File: app/components/pipeline-options/template.hbs [hunk 3]
```diff
@@ -186,6 +186,31 @@
     </section>
   </div>
 
+  <div class="col-xs-12 col-md-8">
+    <section class="preference">
+      <h3>Pipeline Preference</h3>
```
Comment (tkyi) [inline app/components/pipeline-options/template.hbs:191]: not sure if `Pipeline Display Preference` would be more clear or something like that

Comment (adong) [reply app/components/pipeline-options/template.hbs:191]: That "Pipeline Preference" is corresponding to "User Preference".

File: app/components/pipeline-options/template.hbs (hunk 2 above)
Comment (adong) [reply app/components/pipeline-options/template.hbs:197]: image

File: app/components/pipeline-events-list/component.js [hunk 4]
```diff
@@ -7,7 +7,28 @@ import moment from 'moment';
...
 export default Component.extend({
   router: service(),
+  shuttle: service(),
   errorMessage: '',
+  isGroupedEvents: computed('pipeline', {
+    get() {
+      const isGroupedEvents = getWithDefault(
+        this,
+        'pipeline.settings.groupedEvents',
+        false
```
Comment (jithine) [inline app/components/pipeline-events-list/component.js:17]: default should be `true` current behavior

Comment (adong) [reply app/components/pipeline-events-list/component.js]: Updated

File: app/pipeline/model.js [hunk 5]
```diff
@@ -22,7 +22,13 @@ export default DS.Model.extend({
   secrets: DS.hasMany('secret', { async: true }),
   tokens: DS.hasMany('token', { async: true }),
   metrics: DS.hasMany('metric', { async: true }),
-  settings: DS.attr(),
+  settings: DS.attr({
+    defaultValue() {
```
Comment (adong) [inline app/pipeline/model.js:26]: Add default value for `groupedEvents` here

**Review by jithine — APPROVED** [review]

File: app/components/pipeline-options/template.hbs (hunk 3 above)
Comment (tkyi) [reply app/components/pipeline-options/template.hbs:192]: okay

**Pull request merged**